### PR TITLE
closes #35 Add RANDOMIZATION records to DS

### LINKED ISSUE #35: Add RANDOMIZATION records to DS
Add new dummy disposition event records with DSTERM and DSDECOD = "RANDOMIZED" for each patient and for dates it's fine to use those from DM.RFSTDTC.

We need some dummy data here to feed through to admiral so that we can create an ADSL.RANDDT which can in turn be used for admiralonco and further TA-specific package extensions where things like time to event endpoints often are calculated using randomization date.

### PR DESCRIPTION
**Pull request by Gopi2345**
DS is updated by adding records for RANDOMIZED.  Please confirm that the script is reading the correct DS and SUPPDS as inputs.
closes #35

### CODE REVIEW

File: R/ds.R [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#' A SDTM DS dataset from the CDISC pilot project
+#'
+#' @source \url{https://bitbucket.cdisc.org/projects/CED/repos/sdtm-adam-pilot-project/browse/updated-pilot-submission-package/900172/m5/datasets/cdiscpilot01/tabulations/sdtm/ds.xpt}
+#' "ds"
+
+library(dplyr)
+library(labelled)
+library(tidyselect)
+library(admiral)
+library(admiral.test)
```
Comment (rossfarrugia) [inline R/ds.R:14]: please remove the dot - we changed our naming convention so this package is admiraltest now

Comment (rossfarrugia) [inline R/ds.R:14]: please remove the dot - we changed our naming convention so this package is admiraltest now

**Review by rossfarrugia — CHANGES_REQUESTED** [review]

**Comment by rossfarrugia** [comment]
@bundfussr & @yangc71 appreciate if one of you could do a review of this admiraltest update from Gopi, that we need for admiralonco. Fine to do this sometime next week if busy with the admiral release today. Hope you agree with the advice i gave on the issue for this https://github.com/pharmaverse/admiraltest/issues/35. This is the first time we've taken a dataset from the CDISC test pilot and tried to add to it.

**Comment by bms63** [comment]
I think we need this before admiral CRAN release as it is a dependent package? @thomas-neitmann

File: R/ds.R [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+set.seed(1)
+
+# Reading input data
+data("dm")
```
Comment (bundfussr) [inline R/ds.R:19]: Should be `"admiral_dm"` instead of `"dm"`.

File: R/ds.R [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+data("suppdm")
+
+data("ds")
```
Comment (bundfussr) [inline R/ds.R:22]: Should be `"admiral_ds"` instead of `"ds"`.

File: R/ds.R [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+# Converting blank to NA
+dm <- convert_blanks_to_na(dm)
+suppdm <- convert_blanks_to_na(suppdm)
+ds1a <- convert_blanks_to_na(ds)
+suppds1a <- convert_blanks_to_na(suppds)
+
+# Creating full DS data
+ds1 <- ds1a %>%
+  derive_vars_suppqual(suppds1a)
```
Comment (bundfussr) [inline R/ds.R:33]: `derive_vars_suppqual()` is deprecated.

File: R/ds.R [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+library(labelled)
+library(tidyselect)
+library(admiral)
+library(admiral.test)
+
+set.seed(1)
+
+# Reading input data
+data("dm")
+data("suppdm")
```
Comment (bundfussr) [inline R/ds.R:20]: Do we need this?

**Review by bundfussr — CHANGES_REQUESTED** [review]
We have the `ds.R` script twice in the root folder and the R folder.

As it is set up at the moment the script can be run only once because it reads in ds and writes ds. I think we need a guidance how existing dataset should be updated.

We could rename the original data to `raw_ds` and then the script reads in `raw_ds` and writes `admiral_ds`. 
The script is not user facing. Therefore I would put it into the `inst/data_scripts` folder. I would name the script `update_ds.R`. Then it is clear that it updates the raw dataset instead of creating it from the scratch like `tr.R`.

@thomas-neitmann , @rossfarrugia , @bms63 , what do you think?

**Comment by rossfarrugia** [comment]
@bms63 it's fine if this PR comes in next release, so no pressure to merge this today for the admiral release. we would need to do a release for admiraltest though for CRAN due to all the rename recent updates.

@bundfussr - i thought it would be ok if we read in as `ds` and write out `admiral_ds` but happy for @thomas-neitmann & @bms63 to make a call on what's best strategy, as first time we're having this scenario here.

**Comment by bms63** [comment]
> We have the `ds.R` script twice in the root folder and the R folder.
> 
> As it is set up at the moment the script can be run only once because it reads in ds and writes ds. I think we need a guidance how existing dataset should be updated.
> 

I agree we need some guidance on for scripts for updating/reproducing the datasets.  I'd like to make an update for lb in the future and I am sure we will have others.

> We could rename the original data to `raw_ds` and then the script reads in `raw_ds` and writes `admiral_ds`. The script is not user facing. Therefore I would put it into the `inst/data_scripts` folder. I would name the script `update_ds.R`. Then it is clear that it updates the raw dataset instead of creating it from the scratch like `tr.R`.
> 
> @thomas-neitmann , @rossfarrugia , @bms63 , what do you think?

I think this is a good approach

File: R/ds.R (hunk 1 above)
Comment (Gopi2345) [reply R/ds.R]: Updated

Comment (Gopi2345) [reply R/ds.R]: Updated

File: R/ds.R (hunk 2 above)
Comment (Gopi2345) [reply R/ds.R]: updated

File: R/ds.R (hunk 3 above)
Comment (Gopi2345) [reply R/ds.R]: updated

File: R/ds.R (hunk 5 above)
Comment (Gopi2345) [reply R/ds.R]: updated

**Comment by Gopi2345** [comment]
> 

Input data name is changed to raw_ds and raw_suppds and the program name is changed to update_ds.R. All comments are fixed.

File: inst/data_scripts/update_ds.R [hunk 6]
```diff
@@ -0,0 +1,100 @@
+# Update DS by adding DSDECOD=RANDOMIZED rows
+
+library(dplyr)
+library(tidyselect)
+library(labelled)
+library(admiral)
+library(metatools)
+library(admiraltest)
```
Comment (bundfussr) [inline inst/data_scripts/update_ds.R:8]: Sorry, you need to revert it to `admiral.test`.

CRAN rejected renaming `admiral.test` to `admiraltest`.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: inst/data_scripts/update_ds.R (hunk 6 above)
Comment (rossfarrugia) [reply inst/data_scripts/update_ds.R]: sorry @Gopi2345 for the back and forth - i just made this update on your behalf and committed so no action for you for this one

**Review by bundfussr — APPROVED** [review]

**Comment by rossfarrugia** [comment]
done & thanks once again @Gopi2345 for all your efforts!

**Comment by Gopi2345** [comment]
Thank you Ross and Stefan for your support.

**Pull request merged**